epytope/Core/Result.py

```diff
@@ -56,46 +61,61 @@ class EpitopePredictionResult(AResult):
+        |  Method        |    Method 1   |    Method 2   |    Method 1   |    Method 2   |
+        +- - - - - - - - +- - - -+- - - -+- - - -+- - - -+- - - -+- - - -+- - - -+- - - -+
+        | ScoreType      | Score |  Rank | Score |  Rank | Score |  Rank | Score |  Rank |
+        +- - - - - - - - +- - - -+- - - -+- - - -+- - - -+- - - -+- - - -+- - - -+- - - -+
+        |  Peptides      |       |       |       |       |       |       |       |       |
+        +================+=======+=======+=======+=======+=======+=======+=======+=======+
+        | Peptide Obj 1  |  0.03 |  57.4 |  0.05 |  51.1 |  0.08 |  49.4 |  0.73 |  3.12 |
+        +----------------+-------+-------+-------+-------+-------+-------+-------+-------+
+        | Peptide Obj 2  |  0.32 |  13.2 |  0.31 |  14.1 |  0.25 |  22.1 |  0.11 |  69.1 |
+        +----------------+-------+-------+-------+-------+-------+-------+-------+-------+
 
     """
 
-    def filter_result(self, expressions):
+    def filter_result(self, expressions, scoretype=False):
         """
         Filters a result data frame based on a specified expression consisting of a list of triple with
-        (method_name, comparator, threshold). The expression is applied to each row. If any of the columns fulfill
-        the criteria the row remains.
+        (method_name/scoretype_name, comparator, threshold) and a boolean if the scoretype is specified.
+        The expression is applied to each row. If any of the columns fulfill the criteria the row remains.
+
+        :param list((str,comparator,float)) expression: A list of triples consisting of (method_name/scoretype_name, comparator, threshold)
+        :param boolean scoretype: Indicates weather a score or method is specified as first item of the triples in the expressions
 
-        :param list((str,comparator,float)) expressions: A list of triples consisting of (method_name, comparator,
-                                                        threshold)
         :return: Filtered result object
         :rtype: :class:`~epytope.Core.Result.EpitopePredictionResult`
         """
         if isinstance(expressions, tuple):
             expressions = [expressions]
+            
+        df = deepcopy(self)
+        scoretypes = df.columns.get_level_values(2)
+        lvl = 2 if scoretype else 1
+        
+        if scoretype and len(set(scoretypes)) > 1:
+            raise ValueError("Cannot filter dataframe on non-unique ScoreTypes of dataframe.")
+
+        for expr in expressions:
+            key, comp, thr = expr
+            filter = comp(df.xs(key, axis = 1, level = lvl), thr).values
+            if scoretype:
+                if key not in list(set(scoretypes))[0]:
+                    raise ValueError("Specified ScoreType {} does not match ScoreType of Dataframe {}.".format(key, scoretypes[1]))
+                else:
+                    # Only keep rows which contain values above the threshold
```

Comment:
I would change the wording of the comment here and below, since it is not necessarily "above the threshold" but depends on the given comparator.